### PR TITLE
Indexer undertaker

### PR DESCRIPTION
**Pull request by rybmat**
Undertaker worker for indexing pipeline

### CODE REVIEW

File: workers/indexer-undertaker/src/main/resources/banner.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+                            @___@`    \/_   //    |   \   \         \/\ \
+                           0/0/|       \/_ //     |    \    \         \  \
+                       0/0/0/0/|        \///      |     \     \       |  |
+                    0/0/0/0/0/_|_ /   (  //       |      \     _\     |  /
+                 0/0/0/0/0/0/`/,_ _ _/  ) ; -.    |    _ _\.-~       /   /
+                             ,-}        _      *-.|.-~-.           .~    ~
+            \     \__/        `/\      /                 ~-. _ .-~      /
+             \____(oo)           *.   }            {                   /
+             (    (--)          .----~-.\        \-`                 .~
+             //__\\  \__ Ack!   ///.----..<        \             _ -~
```
Comment (SebastianMarzjan) [inline workers/indexer-undertaker/src/main/resources/banner.txt]: Ack! Can we get rid of that? :)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Override
+  public boolean handle(String message, QueueingConsumer.Delivery delivery) {
+    String xdeath = delivery.getProperties().getHeaders().get("x-death").toString();
+    LOGGER.info(getLoggerMarkers(headersToJSON(xdeath)), message);
+    return true;
+  }
+
+  private JSONObject headersToJSON(String headers) {
+    String jsonArray = headers.replaceAll("=", ":")
```
Comment (SebastianMarzjan) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: Please consider having a hashmap for replacements

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+
+public class IndexerUndertaker extends AbstractWorker {
+
+  @Override
+  public boolean handle(String message, QueueingConsumer.Delivery delivery) {
+    String xdeath = delivery.getProperties().getHeaders().get("x-death").toString();
```
Comment (SebastianMarzjan) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: It's worth to extract this string literal to constant in parent class IMO

File: workers/indexer-undertaker/src/main/resources/banner.txt (hunk 1 above)
Comment (rybmat) [inline workers/indexer-undertaker/src/main/resources/banner.txt]: do you mean get rid at all or just change it to something else?

Comment (SebastianMarzjan) [inline workers/indexer-undertaker/src/main/resources/banner.txt]: Well, I see no value in this file :)

Comment (rybmat) [inline workers/indexer-undertaker/src/main/resources/banner.txt]: don't tell me, tell this to services team :p

Comment (gabrys) [inline workers/indexer-undertaker/src/main/resources/banner.txt]: :dragon_face: :dragon: :fire:

File: workers/templates/src/main/java/com/wikia/workers/ExampleWorker.java [hunk 4]
```diff
@@ -5,9 +5,6 @@
 import com.rabbitmq.client.QueueingConsumer;
```
Comment (SebastianMarzjan) [inline workers/templates/src/main/java/com/wikia/workers/ExampleWorker.java]: Is this and following files needed?

Comment (rybmat) [inline workers/templates/src/main/java/com/wikia/workers/ExampleWorker.java]: these are templates for gradle task "newWorker"

File: workers/indexer-undertaker/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# IndexerUndertaker
+catches all rejected messages and sends them to kibana
```
Comment (SebastianMarzjan) [inline workers/indexer-undertaker/README.md]: Please describe the filter to use to access this (for simplicity!)

**Comment by SebastianMarzjan** [comment]
One last comment and if Sonar and tests are OK LGTM

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+   * converts x-death headers to json format
+   * example headers:
+   * [{reason=rejected, count=1, exchange=, time=Tue Oct 13 08:51:43 CEST 2015, routing-keys=[DeleteWorker.article.onArticleDeleteComplete], queue=DeleteWorker.article.onArticleDeleteComplete}]
+   *
+   * @param headers
+   * @return JSONObject
+   */
+  public JSONObject headersToJSON(String headers) {
+    String jsonArray = headers.replaceAll("=", ":")
+        .replaceAll(":,", ":null,")     // sometimes there are keys without values
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        .replaceAll("time:[^,]*,", "")  // time is hard to parse
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        .replaceAll(":", ":\"")         // treat all values as strings
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        .replaceAll(",", "\",")         // treat all values as strings
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        .replaceAll("}", "\"}");        // closing "
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+    return new JSONObject(jsonArray.substring(1, jsonArray.length() - 1)); // removing first [ and last ]
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MINOR 6 minor

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MINOR 6 minor

Watch the comments in this conversation to review them.
